zulip_bots/zulip_bots/bots/idonethis/idonethis.py

```diff
@@ -0,0 +1,212 @@
+
+class UnspecifiedProblemExcept(Exception):
+    pass
+
+def makeAPIRequest(endpoint: str, method: str="GET", body: Dict[str, str]=None) -> Any:
+    global api_key
+    headers = {'Authorization': 'Token ' + api_key}
+    if method == "GET":
+        r = requests.get(API_BASE_URL + endpoint, headers=headers)
+    elif method == "POST":
+        r = requests.post(API_BASE_URL + endpoint, headers=headers, json=body)
+    if r.status_code == 200:
+        return r.json()
+    elif r.status_code == 401 and 'error' in r.json() and r.json()['error']  == "Invalid API Authentication":
+        logging.error('Error authenticating, please check key ' + str(r.url))
+        raise AuthenticationException()
+    else:
+        logging.error('Error make API request, code ' + str(r.status_code) + '. json: ' + r.json())
+        raise UnspecifiedProblemExcept()
+
+def apiNoop() -> Any:
+    return makeAPIRequest("/noop")
+
+def apiListTeam() -> Any:
+    return makeAPIRequest("/teams")
+
+def apiShowTeam(hash_id: str) -> Any:
+    return makeAPIRequest("/teams/{}".format(hash_id))
+
+def apiShowUsers(hash_id: str) -> Any:
+    return makeAPIRequest("/teams/{}/members".format(hash_id))
+
+def apiListEntries(team_id: str=None) -> Any:
+    if team_id:
+        return makeAPIRequest("/entries?team_id={}".format(team_id))
+    else:
+        return makeAPIRequest("/entries".format(team_id))
+
+def apiCreateEntry(body: str, team_id: str) -> Any:
+    return makeAPIRequest("/entries", "POST", {"body": body, "team_id": team_id})
+
+def listTeams() -> str:
+    data = apiListTeam()
+    response = "Teams:\n"
+    for team in data:
+        response += " * " + team['name'] + "\n"
+    return response
+
+def getTeamHash(team_name: str) -> str:
+    data = apiListTeam()
+    for team in data:
+        if team['name'].lower() == team_name.lower() or team['hash_id'] == team_name:
+            return team['hash_id']
+    else:
+        raise TeamNotFoundException(team_name)
+
+def teamInfo(team_name: str) -> str:
+    data = apiShowTeam(getTeamHash(team_name))
+    response = "Team Name: " + data['name'] + "\n"
+    response += "ID: `" + data['hash_id'] + "`\n"
+    response += "Created at: " + data['created_at'] + "\n"
+    return response
+
+def entriesList(team_name: str) -> str:
+    if team_name:
+        data = apiListEntries(getTeamHash(team_name))
+    else:
+        data = apiListEntries()
+    response = "Entries for " + team_name + ":\n"
+    for entry in data:
+        response += " * " + entry['body_formatted'] + "\n"
+        response += "  * Created at: " + entry['created_at'] + "\n"
+        response += "  * Status: " + entry['status'] + "\n"
+        response += "  * User: " + entry['user']['full_name'] + "\n"
+        response += "  * Team: " + entry['team']['name'] + "\n"
+        response += "  * ID: " + entry['hash_id'] + "\n"
+    return response
+
+def createEntry(message: str) -> str:
+    SINGLE_WORD_REGEX = re.compile("--team=([a-zA-Z0-9_]*)")
```

Comment:
Okay, so camelCase naming for functions isn't pep8 compliant (if I remember correctly). Please change all function names to use underscores, for example "create_entry" :)